lib/fluent/compat/output.rb

```diff
@@ -0,0 +1,548 @@
+        emit(tag, es, NULL_OUTPUT_CHAIN)
+      end
+    end
+
+    class MultiOutput < Output
+    end
+
+    class BufferedOutput < Fluent::Plugin::Output
+      # TODO: warn when deprecated
+
+      helpers :event_emitter
+
+      def support_in_v12_style?(feature)
+        case feature
+        when :synchronous    then false
+        when :buffered       then true
+        when :delayed_commit then false
+        when :custom_format  then true
+        end
+      end
+
+      desc 'The buffer type (memory, file)'
+      config_param :buffer_type, :string, default: 'memory'
+      desc 'The interval between data flushes.'
+      config_param :flush_interval, :time, default: 60
+      config_param :try_flush_interval, :float, default: 1
+      desc 'If true, the value of `retry_value` is ignored and there is no limit'
+      config_param :disable_retry_limit, :bool, default: false
+      desc 'The limit on the number of retries before buffered data is discarded'
+      config_param :retry_limit, :integer, default: 17
+      desc 'The initial intervals between write retries.'
+      config_param :retry_wait, :time, default: 1.0
+      desc 'The maximum intervals between write retries.'
+      config_param :max_retry_wait, :time, default: nil
+      desc 'The number of threads to flush the buffer.'
+      config_param :num_threads, :integer, default: 1
+      desc 'The interval between data flushes for queued chunk.'
+      config_param :queued_chunk_flush_interval, :time, default: 1
+
+      desc 'The size of each buffer chunk.'
+      config_param :buffer_chunk_limit, :size, default: 8*1024*1024
+      desc 'The length limit of the chunk queue.'
+      config_param :buffer_queue_limit, :integer, default: 256
+      desc 'The action when the size of buffer queue exceeds the buffer_queue_limit.'
+      config_param :buffer_queue_full_action, :enum, list: [:exception, :block], default: :exception
+
+      config_param :flush_at_shutdown, :bool, default: true
+
+      PARAMS_MAP = [
+        ["buffer_type", "@type"],
+        ["num_threads",    "flush_threads"],
+        ["flush_interval", "flush_interval"],
+        ["try_flush_interval",          "flush_thread_interval"],
+        ["queued_chunk_flush_interval", "flush_burst_interval"],
+        ["disable_retry_limit",         "retry_forever"],
+        ["retry_limit",                 "retry_max_times"],
+        ["max_retry_wait",              "retry_max_interval"],
+        ["buffer_chunk_limit", "chunk_bytes_limit"],
+        ["buffer_queue_limit", "queue_length_limit"],
+        ["buffer_queue_full_action", nil], # TODO: implement this on fluent/plugin/buffer
+        ["flush_at_shutdown",  "flush_at_shutdown"],
+      ]
+
+      def configure(conf)
+        bufconf = CompatOutputUtils.buffer_section(conf)
+        config_style = (bufconf ? :v1 : :v0)
+        if config_style == :v0
+          buf_params = {
+            "flush_mode" => "fast",
+            "retry_type" => "expbackoff",
```

Comment:
'expbackoff' is too short for me. 'exponential_backoff' is better.
Maybe, almost users don't change this parameter so verbose is better.